REGRESSION (Sonoma 14.1): [ Sonoma+ Debug arm64 ] TestWebKitAPI.ContextMenuTests.ContextMenuElementInfoWithQRCodeDetectionCrash is a flaky failure

martadarbinyan opened the pull request:
#### 59a2351367374efac283685f5a0acbd2aa876d7a
<pre>
REGRESSION (Sonoma 14.1): [ Sonoma+ Debug arm64 ] TestWebKitAPI.ContextMenuTests.ContextMenuElementInfoWithQRCodeDetectionCrash is a flaky failure
<a href="https://rdar.apple.com/111706229">rdar://111706229</a>

Reviewed by Ryan Haddad.

Disabling failing API test.

* Tools/TestWebKitAPI/Tests/mac/ContextMenuTests.mm:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/272435@main">https://commits.webkit.org/272435@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b5a0a49ba786652946196d2e1dd5ebc47e0ba7ca

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31585 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/10263 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/33299 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/34077 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/28609 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/32362 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12617 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7514 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/28217 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31933 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8647 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/28193 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7451 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7617 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/28100 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/35422 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28712 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28550 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33740 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7700 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5714 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/31606 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/9355 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8385 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4131 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8206 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ryanhaddad commented on Tools/TestWebKitAPI/Tests/mac/ContextMenuTests.mm:
```diff
@@ -468,7 +468,12 @@ static void swizzledOrderFrontColorPanel(id, SEL, id)
 }
 
+// FIXME: Re-enable this test once webkit.org/b/266650 is resolved
+#if PLATFORM(MAC) && (__MAC_OS_X_VERSION_MIN_REQUIRED > 140000) && !defined(NDEBUG)
```
style nit: you don't need the parentheses around `__MAC_OS_X_VERSION_MIN_REQUIRED`

martadarbinyan committed:
REGRESSION (Sonoma 14.1): [ Sonoma+ Debug arm64 ] TestWebKitAPI.ContextMenuTests.ContextMenuElementInfoWithQRCodeDetectionCrash is a flaky failure
rdar://111706229

Reviewed by Ryan Haddad.

Disabling failing API test.

* Tools/TestWebKitAPI/Tests/mac/ContextMenuTests.mm:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/272435@main
